10-05/task_7/task7.rb

```diff
@@ -0,0 +1,38 @@
+require 'csv'
+
+CSV.open("visit_stat.csv", "w") do |wr|
+  wr <<  ["IP", "Day", "Time"]
+  wr <<  ["139.18.150.126", "Friday", "23:12:44"]
+  wr <<  ["139.18.150.126", "Monday", "21:10:30"]
+  wr <<  ["165.11.365.112", "Friday", "21:15:35"]
+  wr <<  ["139.18.150.126", "Wednesday", "12:12:10"]
+  wr <<  ["197.33.150.432", "Friday", "10:50:44"]
+  wr <<  ["167.54.188.114", "Tuesday", "21:47:40"]
+  wr <<  ["162.57.188.115", "Tuesday", "09:50:40"]
+end
+
+sorted_ip = ""
+
+visit_list = CSV.read("visit_stat.csv", headers: true)
+
+ip_hash_counter = Hash.new(0)
+visit_list["IP"].each { |ip| ip_hash_counter[ip] += 1 }
+ip_hash_counter.each do |ip, number|
+  sorted_ip << "#{ip} - #{number} \n"
+end
+
+day_hash_counter = Hash.new(0)
+visit_list["Day"].each { |day| day_hash_counter[day] += 1 }
+day_counter = day_hash_counter.sort{|a,b| b[1]<=>a[1]}
```

Comment:
`sort` is a bit expensive operation and will take a lot of time on big amounts of data.
It's better to use `#max_by` here so you can get exactly what you need here.
FYI: here is a comparison between all ruby `max` methods available https://dev.to/sylviapap/the-ruby-enumerable-operator-and-max-vs-maxby-ac6